File: PTA/PAT_A/Python3/A1108_AC.py
def is_legal(t: str) -> bool:
    def output(ok: bool) -> bool:
        if not ok:
            print('ERROR: %s is not a legal number' % t)
        return ok

    try:
        num = float(t)
    except:
        return output(False)

    if num >= -1000 and num <= 1000:
        dot_i = t.find('.')
        if dot_i != -1:
            return output(len(t)-1-dot_i <= 2)
        else:
            return output(True)
    else:
        return output(False)


_ = input()
suspicious = input().split()

numbers = list(map(float, filter(is_legal, suspicious)))

cnt = len(numbers)

if cnt >= 2:
    print('The average of %d numbers is %.2f' % (cnt, sum(numbers)/cnt))
elif cnt == 1:
    print('The average of 1 number is %.2f' % numbers[0])
else:
    print('The average of 0 numbers is Undefined')

File: PTA/PAT_A/Cpp11/A1108_AC.cc
// ---
// id         : 1108
// title      : Finding Average
// difficulty : Easy
// score      : 20
// tag        : TODO
// keyword    : TODO
// status     : AC
// from       : PAT (Advanced Level) Practice
// ---

// reference: https://www.liuchuo.net/archives/1924

#include <cstdio>
#include <cstring>
#include <iostream>
using namespace std;
int main() {
  int n, cnt = 0;
  cin >> n;
  char bufin[50], bufout[50];
  double temp, sum = 0.0;
  for (int i = 0; i < n; i++) {
    scanf("%s", bufin);
    sscanf(bufin, "%lf", &temp);
    sprintf(bufout, "%.2f", temp);
    bool is_valid = true;
    for (int j = 0; j < strlen(bufin); j++)
      if (bufin[j] != bufout[j])
        is_valid = false;
    if (!is_valid || temp < -1000 || temp > 1000) {
      printf("ERROR: %s is not a legal number\n", bufin);
      continue;
    } else {
      sum += temp;
      cnt++;
    }
  }
  if (cnt == 1)
    printf("The average of 1 number is %.2f", sum);
  else if (cnt > 1)
    printf("The average of %d numbers is %.2f", cnt, sum / cnt);
  else
    printf("The average of 0 numbers is Undefined");
  return 0;
}